icu4c/source/common/uloc_tag.cpp

```diff
@@ -1938,6 +2125,37 @@ ultag_parse(const char* tag, int32_t tagLen, int32_t* parsedLen, UErrorCode* sta
         }
     }
 
+    size_t parsedLenDelta = 0;
+    if (grandfatheredLen == 0) {
+        for (i = 0; REDUNDANT[i] != nullptr; i += 2) {
```

Comment:
Instead of depending on this list to be null-terminated, I suggest that you use `UPRV_LENGTHOF()` for the loop condition, like you already do for the other lists.

Reply:
Done. Thanks